php/libraries/NDB_Config.class.inc

```diff
@@ -570,9 +572,7 @@ class NDB_Config
             $nextLevel = NDB_Config::getMenuTabs($thisRow['ID']);
 
             if (!empty($nextLevel)) {
-                if (NDB_Config::checkMenuPermission($nextLevel[0]['ID'])) {
```

Comment:
I don't think that's accurate.

The check on 559 looks like it's checking if any children of a level have permission or if the whole level should be excluded.

This check is while going through the individual rows, to ensure that the user has permission for each individual row to exclude the ones that are wrong.